### ISSUE TITLE
ValueError: Length mismatch: Expected axis has 0 elements, new values have 5 elements

### ISSUE THREAD

**Issue by VincentDuf** [sammchardy/python-binance#1122]
Hi,
I'm trying to get the daily data from all BUSD and USDT pairs from Binance over the past 2 years.

Here is the main function I use to get these data:

```
def klines_2_dataframe(ticker):
    new_df = pd.DataFrame(client.get_historical_klines(ticker,"1d","1 Jan, 2020"))
    new_df = new_df.iloc[:,:5]
    new_df.columns = ['Time','open','high','low','close']
    new_df[["open","high","low","close"]] = new_df[["open","high","low","close"]].astype(float)
    new_df.Time = pd.to_datetime(new_df.Time,unit="ms")
    return new_df
```

And how I want to save those dataframe inside a SQLite database:

```
db = sqlite3.connect(os.path.join(os.getcwd(),"busd_pairs.sqlite"))
for tickers in busd_symbols:
   df = klines_2_dataframe(tickers)
   df.to_sql(tickers,db,index=False,index_label=df.Time,if_exists='replace')
```

Weirdly when I execute this line alone:

`new_df = pd.DataFrame(client.get_historical_klines(ticker,"1d","1 Jan, 2020"))`

There is no error. But when I run it inside a loop I get this error over and over:
`ValueError: Length mismatch: Expected axis has 0 elements, new values have 5 elements`

Despite some research on stackoverflow or other site I still don't get the error... if anyone can help me.

Thanks :)

**Comment by VincentDuf** [comment]
No one ? @sammchardy 
Even when I pre-create a DataFrame with 5 empty columns (or with one row) I still get the error... is it coming from the binance side or is it pandas ? 

Thanks

**Comment by katkoe** [comment]
For the pairs that throw up this error no data is returned by the api, so the dataframe is empty and therefore has 0 columns instead of 5. I'm not sure what causes this.

**Comment by katkoe** [comment]
Not sure if this is too late, but I'm pretty sure this error is returned for delisted pairs. The Binance api returns an empty dict for these pairs, so the dataframe is empty in these instances and therefore has 0 columns. When you try to give 5 column names (in new_df.columns = ['Time','open','high','low','close']) for a 0 column dataframe Pandas throws up this error.

**Comment by Simmelariii** [comment]
> katkoe
This is not true, i get the same error after looping through 5-100 times using only btcusdt 
<img width="1440" alt="Skärmavbild 2022-07-10 kl  00 52 13" src="https://user-images.githubusercontent.com/108994392/178125092-952e40c6-e9c2-4331-b73d-182b2fc4bf7f.png">
If you know another soulution i would love to hear it, thanks!

**Comment by Dhruv-16** [comment]
Has anyone solved this yet? 
I have got a same error:

Output exceeds the [size limit](command:workbench.action.openSettings?[). Open the full output data [in a text editor](command:workbench.action.openLargeOutput?4bcf9bbe-13d8-4c50-9fdb-7d85b12cae7c)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
c:\Users\dhruv\Documents\TRADING\Lakshmi.ipynb Cell 5 in <cell line: 11>()
      [8](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=7)     frame = frmae.astype(float)
      [9](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=8)     return frame
---> [11](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=10) getminutedata(asset , interval , lookback)

c:\Users\dhruv\Documents\TRADING\Lakshmi.ipynb Cell 5 in getminutedata(asset, interval, lookback)
      [3](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=2) frame = pd.DataFrame(Data)
      [4](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=3) frame = frame.iloc[:,:6]
----> [5](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=4) frame.columns = ['Time' , 'Open' , 'High', 'Low' , 'Close' , 'Volume']
      [6](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=5) frame = frame.set_index('Time')
      [7](vscode-notebook-cell:/c%3A/Users/dhruv/Documents/TRADING/Lakshmi.ipynb#W6sZmlsZQ%3D%3D?line=6) frame.index = pd.to_datetime(frame , unit='ms')

File c:\Users\dhruv\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\generic.py:5588, in NDFrame.__setattr__(self, name, value)
   5586 try:
   5587     object.__getattribute__(self, name)
-> 5588     return object.__setattr__(self, name, value)
   5589 except AttributeError:
   5590     pass

File c:\Users\dhruv\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\_libs\properties.pyx:70, in pandas._libs.properties.AxisProperty.__set__()

File c:\Users\dhruv\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\core\generic.py:769, in NDFrame._set_axis(self, axis, labels)
    767 def _set_axis(self, axis: int, labels: Index) -> None:
...
     70         f"Length mismatch: Expected axis has {old_len} elements, new "
     71         f"values have {new_len} elements"
     72     )

ValueError: Length mismatch: Expected axis has 0 elements, new values have 6 elements


my code is:
 def getminutedata(asset , interval , lookback):
    Data = client.get_historical_klines(asset , '1m' , '120m')
    frame = pd.DataFrame(Data)
    frame = frame.iloc[:,:6]
    frame.columns = ['Time' , 'Open' , 'High', 'Low' , 'Close' , 'Volume']
    frame = frame.set_index('Time')
    frame.index = pd.to_datetime(frame , unit='ms')
    frame = frmae.astype(float)
    return frame

getminutedata(asset , interval , lookback)

asset,interval,lookback are global variable so no issue with that just that i think there is no stream of datasets to pd from binance api even though the api is up and running i tested the api by calling (info = client.get_account())
